tmt/utils/templates.py

```diff
@@ -0,0 +1,362 @@
+"""
+Template rendering.
+
+Package provides primitives for Jinja2 template rendering, plus our
+custom filters.
+"""
+
+import re
+import textwrap
+from re import Match
+from typing import (
+    Any,
+    Callable,
+    Optional,
+    cast,
+    )
+
+import fmf
+import fmf.utils
+import jinja2
+import jinja2.exceptions
+
+from tmt.utils import GeneralError, Path
+
+
+def _template_filter_basename(  # type: ignore[reportUnusedFunction,unused-ignore]
+        pathlike: str) -> str:
+    """
+    Return the last component of the given path.
+
+    .. code-block:: jinja
+
+        # "/etc/fstab" -> "fstab"
+        {{ "/etc/fstab" | basename }}
+
+        # "/var/log/" -> "log"
+        {{ "/var/log/" | basename }}
+    """
+
+    return Path(pathlike).name
+
+
+def _template_filter_match(  # type: ignore[reportUnusedFunction,unused-ignore]
+        s: str, pattern: str) -> Optional[Match[str]]:
```

Comment:
Is it `Match[str]`, when returning the whole Match object?

Reply:
I'd say it is, `Match` is a generic type, it can be also `Match[bytes]` when working with `bytes` instead of `str`ings.

Reply:
@martinhoyer is it possible to resolve this conversation?